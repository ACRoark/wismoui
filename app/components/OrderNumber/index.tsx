import * as React from 'react';

interface IOrderNumberProps {
  value: string;
}

const OrderNumber: React.FC<IOrderNumberProps> = (props: IOrderNumberProps): React.ReactElement => {
  const { value } = props;

  const formatOrderNumber = (input: string): string => {
    return `${input.substr(0, 2)} ${input.substr(2, 4)} ${input.substr(6, 10)}`;
  };

  const regex = new RegExp('^[A-Z]{2}[0-9]{8}$', 'g');

  if (!regex.test(value)) {
    throw new Error('orders must start with MI or MO or MT followed by 8 digits');
  }

  return <span>{formatOrderNumber(value)}</span>;
};

export default OrderNumber;
