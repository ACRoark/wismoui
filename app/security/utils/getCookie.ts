const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
