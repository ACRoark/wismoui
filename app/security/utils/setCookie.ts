interface ICookieOptions {
  days: number;
  path: string;
}

const setCookie = (name: string, value: string, options: ICookieOptions): void => {
  const expires = new Date(Date.now() + options.days * 864e5).toUTCString();

  document.cookie = `${name}=${encodeURIComponent(value)}; expires=${expires}; path=${options.path}`;
};

export default setCookie;
