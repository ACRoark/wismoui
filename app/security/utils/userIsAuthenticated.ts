import useAuth from 'hooks/useAuth';

const userIsAuthenticated = (): boolean => {
  const { isAuthenticated } = useAuth();

  return isAuthenticated;
};

export default userIsAuthenticated;
