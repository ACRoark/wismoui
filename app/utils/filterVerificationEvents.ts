import IVerificationEvent from 'types/IVerificationEvent';
import filterVerificationEventStatusUpdatesByActive from './filterVerificationEventStatusUpdatesByActive';
import filterVerificationEventStatusUpdatesByVerificationCodes from './filterVerificationEventStatusUpdatesByVerificationCodes';

const filterVerificationEvents = (verificationEvents: IVerificationEvent[], verificationCodes: string[]): IVerificationEvent[] => {
  const filteredVerificationEvents: IVerificationEvent[] = [];
  verificationEvents.forEach((event: IVerificationEvent): void => {
    const activeVerificationEventStatusUpdates = filterVerificationEventStatusUpdatesByActive(event.verificationEventStatusUpdates);
    const verificationEventStatusUpdatesByVerificationCodes =
      filterVerificationEventStatusUpdatesByVerificationCodes(activeVerificationEventStatusUpdates, verificationCodes);

    if (verificationEventStatusUpdatesByVerificationCodes.length > 0) {
      const filteredEvent = {
        ...event,
        verificationEventStatusUpdates: verificationEventStatusUpdatesByVerificationCodes
      };

      filteredVerificationEvents.push(filteredEvent);
    }
  });

  return filteredVerificationEvents;
};

export default filterVerificationEvents;
