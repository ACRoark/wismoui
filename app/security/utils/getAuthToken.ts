import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
