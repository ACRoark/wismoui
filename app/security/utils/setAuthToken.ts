import setCookie from './setCookie';

const setAuthToken = (value?: string): void => {
  if (value) {
    setCookie('token', `Bearer ${value}`, { days: 1, path: '/' });
  }
};

export default setAuthToken;
