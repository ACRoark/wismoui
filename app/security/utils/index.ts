import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
