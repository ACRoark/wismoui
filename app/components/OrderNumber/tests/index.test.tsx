import { render, shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import { createSnapshot } from 'testing/utils';

import OrderNumber from '..';

describe('<OrderNumber />', (): void => {
  it('should not log errors in console when rendered', (): void => {
    const spy = jest.spyOn(global.console, 'error');

    render(<OrderNumber value="MI12345678" />);

    expect(spy).not.toHaveBeenCalled();
  });

  it('should render correctly when given a valid value', (): void => {
    const tree = createSnapshot(<OrderNumber value="MI12345678" />);

    expect(tree).toMatchSnapshot();
  });

  it('should throw an error when order number has non-numeric characters', (): void => {
    // tslint:disable-next-line: no-any
    expect((): ShallowWrapper<any> =>  shallow(<OrderNumber value="ABCDEFGHIJKL" />)).toThrowError();
  });

  it('should throw an error when order number is less than 10 characters', (): void => {
    // tslint:disable-next-line: no-any
    expect((): ShallowWrapper<any> => shallow(<OrderNumber value="MO123456"/>)).toThrowError();
  });

  it('should throw an error when order number is more than 10 characters', (): void => {
    // tslint:disable-next-line: no-any
    expect((): ShallowWrapper<any> => shallow(<OrderNumber value="MT1234561234567"/>)).toThrowError();
  });
});
