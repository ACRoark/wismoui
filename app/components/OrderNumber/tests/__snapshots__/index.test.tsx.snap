// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderNumber /> should render correctly when given a valid value 1`] = `
<span>
  MI 1234 5678
</span>
`;
